### PR TITLE
PYTHON-2668 Improve README installation instructions and add FAQ

### PR DESCRIPTION
**Pull request by prashantmital**

### CODE REVIEW

File: bindings/python/README.rst [hunk 1]
```diff
@@ -42,11 +42,23 @@ suited for in-memory analytical processing applications.
...
-PyMongoArrow is available on PyPI:::
+PyMongoArrow is available on PyPI::
 
   $ python -m pip install pymongoarrow
 
-PyMongoArrow wheels are available for macOS and Linux on x86_64 architectures.
+To use PyMongoArrow with MongoDB Atlas' ``mongodb+srv://`` URIs, you will
+need to also install PyMongo with the ``srv`` extra::
+
+  $ python -m pip install pymongo[srv]<4 pymongoarrow
```
Comment (ShaneHarvey) [inline bindings/python/README.rst:52]: Suggest adding quotes to make this copy/pastable in all terminals: `'pymongo[srv]<4'`

File: bindings/python/docs/source/faq.rst [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+.. contents::
+
+Why do I get ``ModuleNotFoundError: No module named 'pandas'`` when using PyMongoArrow
+--------------------------------------------------------------------------------------
+
+This error is raised when an application attempts to use a PyMongoArrow API
+that returns query result sets as a :class:`pandas.DataFrame` instance without
+having ``pandas`` installed in the Python environment. Since ``pandas`` is not
+a direct dependency of PyMongoArrow, it is not automatically installed when
+you install ``pymongoarrow`` and must be installed separately:::
```
Comment (ShaneHarvey) [inline bindings/python/docs/source/faq.rst:13]: There's an extra `:`

File: bindings/python/docs/source/index.rst [hunk 3]
```diff
@@ -21,6 +21,9 @@ know to use **PyMongoArrow**.
 :doc:`supported_types`
   A list of BSON types that are supported by PyMongoArrow.
 
+:doc:`faq`
+  Some questions that come up often.
```
Comment (ShaneHarvey) [inline bindings/python/docs/source/index.rst:25]: "Frequently asked questions"

File: bindings/python/docs/source/installation.rst [hunk 4]
```diff
@@ -58,8 +58,13 @@ For example, to use PyMongoArrow with MongoDB Atlas' ``mongodb+srv://`` URIs
 users must install PyMongo with the ``srv`` extra in addition to installing
 PyMongoArrow::
 
-  $ python -m pip install pymongoarrow
-  $ python -m pip install pymongo[srv]>=3.11,<4
+  $ python -m pip install pymongo[srv]<4 pymongoarrow
```
Comment (ShaneHarvey) [inline bindings/python/docs/source/installation.rst:61]: Same comment about quotes.

File: bindings/python/README.rst [hunk 5]
```diff
@@ -42,11 +42,23 @@ suited for in-memory analytical processing applications.
...
+
+.. note:: To use PyMongoArrow APIs that return query result sets as pandas
```
Comment (ShaneHarvey) [inline bindings/python/README.rst:54]: Remove `.. note::` since it doesn't work in Github.

File: bindings/python/README.rst [hunk 6]
```diff
@@ -42,11 +42,23 @@ suited for in-memory analytical processing applications.
...
+   DataFrame instances, you will also need to have the ``pandas`` package
+   installed::
+
+     $ python -m pip install pandas
+
+.. note:: Currently, PyMongoArrow wheels are only available for macOS and Linux
```
Comment (ShaneHarvey) [inline bindings/python/README.rst:60]: Remove `.. note::` since it doesn't work in Github.

Comment (prashantmital) [reply bindings/python/README.rst:60]: Done.

File: bindings/python/README.rst (hunk 5 above)
Comment (prashantmital) [reply bindings/python/README.rst:54]: Done.

File: bindings/python/README.rst (hunk 1 above)
Comment (prashantmital) [reply bindings/python/README.rst:52]: Done.

File: bindings/python/docs/source/faq.rst (hunk 2 above)
Comment (prashantmital) [reply bindings/python/docs/source/faq.rst:13]: Removed.

File: bindings/python/docs/source/index.rst (hunk 3 above)
Comment (prashantmital) [reply bindings/python/docs/source/index.rst:25]: Changed.

**Review by ShaneHarvey — APPROVED** [review]

**Pull request merged**